code/modules/ai/ai_behaviors/carbon/xeno/xeno.dm

```diff
@@ -16,9 +16,15 @@
 			continue
 		return_result += h
 	var/mob/living/carbon/xenomorph/xeno_parent = mob_parent
-	for(var/x in GLOB.alive_xeno_list)
-		if(!xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will attack as well!
-			return_result += x
+	for(var/x in cheap_get_xenos_near(mob_parent, 8))
+		if(xeno_parent.issamexenohive(x)) //Xenomorphs not in our hive will be attacked as well!
+			continue
+		var/mob/nearby_xeno = x
+		if(nearby_xeno.stat == DEAD)
+			continue
+		if((nearby_xeno.status_flags & GODMODE) || (nearby_xeno.status_flags & GODMODE)) //No attacking the hivemind!
```

Comment:
```suggestion
		if((nearby_xeno.status_flags & INCORPOREAL) || (nearby_xeno.status_flags & INCORPOREAL)) //No attacking the hivemind!
```

Reply:
INCORPOREAL and GODMODE are now both checked